hedera-node/hedera-network-admin-service-impl/src/main/java/com/hedera/node/app/service/networkadmin/impl/handlers/FreezeHandler.java

```diff
@@ -142,25 +141,27 @@ public void handle(@NonNull final HandleContext context) throws HandleException
         // @todo('Issue #6761') - the below switch returns a CompletableFuture, need to use this with an ExecutorService
         switch (freezeTxn.freezeType()) {
             case PREPARE_UPGRADE -> {
-                // by the time we get here, we've already checked that updateFileID is non-null in validateSemantics()
-                // and that fileHash is non-null in preHandle()
-                specialFileStore.updateFileHash(freezeTxn.fileHash());
-                specialFileStore.updateFileID(updateFileID);
-                upgradeActions.extractSoftwareUpgrade(specialFileStore
-                        .get(requireNonNull(updateFileID))
-                        .orElseThrow(() -> new IllegalStateException("Update file not found")));
+                // by the time we get here, we've already checked that fileHash is non-null in preHandle()
+                upgradeStore.updateFileHash(freezeTxn.fileHash());
+                try {
+                    upgradeActions.extractSoftwareUpgrade(upgradeFileStore.getFull());
+                } catch (IOException e) {
+                    throw new IllegalStateException("Error extracting upgrade file", e);
+                }
             }
             case FREEZE_UPGRADE -> upgradeActions.scheduleFreezeUpgradeAt(requireNonNull(freezeStartTimeInstant));
             case FREEZE_ABORT -> {
                 upgradeActions.abortScheduledFreeze();
-                specialFileStore.updateFileHash(null);
-                specialFileStore.updateFileID(null);
+                upgradeStore.updateFileHash(null);
+            }
+            case TELEMETRY_UPGRADE -> {
+                try {
+                    upgradeActions.extractTelemetryUpgrade(
+                            upgradeFileStore.getFull(), requireNonNull(freezeStartTimeInstant));
+                } catch (IOException e) {
+                    throw new IllegalStateException("Error extracting upgrade file", e);
+                }
             }
-            case TELEMETRY_UPGRADE -> upgradeActions.extractTelemetryUpgrade(
-                    specialFileStore
-                            .get(requireNonNull(updateFileID))
-                            .orElseThrow(() -> new IllegalStateException("Telemetry update file not found")),
-                    requireNonNull(freezeStartTimeInstant));
```

Comment:
Should the exception message here be `"Telemetry update file not found"`?

Reply:
I'm confused by your comment. Are you suggesting a different message?

Reply:
The `TELEMETRY_UPGRADE` case had `"Telemetry update file not found"` as the exception message before the change. Now the exception message reads the same as the exception message for `PREPARE_UPGRADE` (L149) that was also changed, `"Error extracting upgrade file"`. Intentional?

Reply:
Yes that was intentional. The circumstances that cause the error have changed slightly, since now we're getting the file from file service instead of the temporary mock I had in place before. We've already checked that file exists in upgradeStore.peek() which is called from preHandle. So if an error occurs here, it's because the file exists but is not a valid zip.